torch/csrc/distributed/c10d/ProcessGroup.hpp

```diff
@@ -151,12 +151,14 @@ class TORCH_API ProcessGroup : public torch::CustomClassHolder {
                     at::TensorList,
                     const c10::intrusive_ptr<::c10d::ProcessGroup>&,
                     const c10::intrusive_ptr<::c10d::ReduceOp>&,
+                    const c10::optional<at::Tensor>& sparse_indices,
```

Comment:
Btw, should `sparse_indices` be explicitly named `output_sparse_indices`? (because if the input `tensor` is already a sparse tensor, you don't need sparse indices for it.)

Reply:
The use of `sparse_indices` is for the case in which users want to override the output tensor with specified sparse indices (as in the case of sharded embedding tables). But we also want to support the use case with just `dist.all_reduce(tensor)` where tensor is a sparse tensor. In this case the `sparse_indices` argument would be empty

Agree that the naming can be updated.